WhateverGreen/kern_igfx.cpp

```diff
@@ -996,11 +996,55 @@ OSObject *IGFX::wrapCopyExistingServices(OSDictionary *matching, IOOptionBits in
 	return FunctionCast(wrapCopyExistingServices, callbackIGFX->orgCopyExistingServices)(matching, inState, options);
 }
 
+bool IGFX::removeVTSupportedProfile2(IOService *that) {
+	if (!that)
+		return false;
+
+	auto hevcDecodeCap = OSDynamicCast(OSDictionary, that->getProperty("IOGVAHEVCDecodeCapabilities"));
+	if (!hevcDecodeCap)
+		return false;
+
+	auto hevcDecodeCapCpy = OSDictionary::withDictionary(hevcDecodeCap);
+	if (!hevcDecodeCapCpy)
+		return false;
+
+	auto vtSuppProf = OSDynamicCast(OSArray, hevcDecodeCapCpy->getObject("VTSupportedProfileArray"));
+	if (!vtSuppProf) {
+		hevcDecodeCapCpy->release();
+		return false;
+	}
+
+	auto vtSuppProfCpy = OSArray::withArray(vtSuppProf);
+	if (!vtSuppProfCpy) {
+		hevcDecodeCapCpy->release();
+		return false;
+	}
+
+	auto count = vtSuppProfCpy->getCount();
+	bool found = false;
+	for (auto i = 0; i < count; i++) {
+		auto num = OSDynamicCast(OSNumber, vtSuppProfCpy->getObject(i));
+		if (num->unsigned8BitValue() == 2) {
+			DBGLOG("igfx", "removing profile 2 from VTSupportedProfileArray/IOGVAHEVCDecodeCapabilities index %u on Skylake with KBL kexts", i);
```

Comment:
Missing null check for num.

Reply:
https://github.com/acidanthera/WhateverGreen/commit/1d26c0f33ff6627e07d563bcc96854bdf8b33dd7